eth/gasestimator/gasestimator.go

```diff
@@ -78,7 +78,6 @@ func Estimate(ctx context.Context, call *core.Message, opts *Options, gasCap uin
 		available := celoBalance
 		if call.FeeCurrency != nil {
 			if !call.IsFeeCurrencyDenominated() {
```

Comment:
I would imagine we can remove this block, since the above check establishes that `FeeCurrency` is not nil, therefore `IsFeeCurrencyDenominated` will always return true, so this block will never be executed.